src/Microsoft.Framework.ClosedGenericMatcher.Sources/ClosedGenericMatcher.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.Framework.Internal
+{
+    /// <summary>
+    /// Helper related to generic interface definitions and implementing classes.
+    /// </summary>
+    internal static class ClosedGenericMatcher
+    {
+        /// <summary>
+        /// Determine whether <paramref name="queryType"/> is or implements a closed generic <see cref="Type"/>
+        /// created from <paramref name="interfaceType"/>.
+        /// </summary>
+        /// <param name="queryType">The <see cref="Type"/> of interest.</param>
+        /// <param name="interfaceType">The open generic <see cref="Type"/> to match. Usually an interface.</param>
+        /// <returns>
+        /// The closed generic <see cref="Type"/> created from <paramref name="interfaceType"/> that
+        /// <paramref name="queryType"/> is or implements. <c>null</c> if the two <see cref="Type"/>s have no such
+        /// relationship.
+        /// </returns>
+        /// <remarks>
+        /// Name could have been <c>ExtractClosedGenericTypeMatchingOpenGenericInterfaceOrSealedType()</c> but that is
+        /// too long. In any case, this method will return <paramref name="queryType"/> if
```

Comment:
The comments about the name don't really add anything, and won't be useful to readers 6 months from now. Just seems like snark :laughing: